scipy/stats/tests/test_multivariate.py

```diff
@@ -2479,3 +2481,107 @@ def test_random_state_property():
         check_pickling(distfn, args)
+
+class TestDirichletMultinomial:
+    def test_covariance_diagonal(self):
+        #Makes sure that the diagonal of the
+        #covariance matrix is the variance.
+        alpha = [1, 5, 3]
+        n = 3
+        c = dirichlet_multinomial.cov(alpha, n)
+        v = dirichlet_multinomial.var(alpha, n)
+
+        assert_array_equal(v, c.diagonal())
+
+    def test_variance(self):
+        alpha = [0.3, 2.1, 5.4]
+        n = 3
+        #Truncated to the 6th digit, so not
+        #exactly what we would expect from var
+        expected_v = [0.136162, 0.724381, 0.784293]
+        v = dirichlet_multinomial.var(alpha, n)
+        assert_array_almost_equal(expected_v, v)
+
+    def test_covariance(self):
+        alpha = [0.3, 2.1, 5.4]
+        n = 3
+        #Truncated to the 6th digit, so not
+        #exactly what we would expect from var
+        expected_c = np.array([[0.136162, -0.038125, -0.098037],
+                               [-0.038125, 0.724381, -0.686256],
+                               [-0.098037, -0.686256, 0.784293]])
+        c = dirichlet_multinomial.cov(alpha, n)
+        assert_array_almost_equal(expected_c, c)
+
+    def test_pmf(self):
+        x = np.array([1, 2, 3])
+        n = 3
+        alpha = np.array([3, 4, 5])
+        y = 2.884615384615385
+        y_1 = dirichlet_multinomial.pmf(alpha, x, n)
+        assert_almost_equal(y, y_1)
+
+    def test_logpmf(self):
+        x = np.array([1, 2, 3])
+        n = 3
+        alpha = np.array([3, 4, 5])
+        y = dirichlet_multinomial.pmf(alpha, x, n)
+        y_1 = dirichlet_multinomial.logpmf(alpha, x, n)
+        assert_almost_equal(y, np.exp(y_1))
+
+    def test_broadcasting(self):
+        x_1 = np.array([1, 2])
+        x_2 = np.array([4, 5])
+        n_1 = 3
+        n_2 = 4
+        alpha_1 = np.array([3, 4])
+        alpha_2 = np.array([6, 7])
+        x = np.array([[1, 2],
+                      [4, 5]])
+
+        alpha = np.array([[3, 4],
+                          [6, 7]])
+
+        n = np.array([3, 4])
+        y = dirichlet_multinomial.logpmf(alpha, x, n)
+        y_1 = dirichlet_multinomial.logpmf(alpha_1, x_1, n_1)
+        y_2 = dirichlet_multinomial.logpmf(alpha_2, x_2, n_2)
+        assert_equal(y[0], y_1)
+        assert_equal(y[1], y_2)
+
+    def test_broadcasting_mean(self):
+        n_1 = 3
+        n_2 = 4
+        alpha_1 = np.array([3, 4])
+        alpha_2 = np.array([6, 7])
+
+        alpha = np.array([[3, 4],
+                          [6, 7]])
+
+        n = np.array([3, 4])
+
+        y = dirichlet_multinomial.mean(alpha, n)
+        y_1 = dirichlet_multinomial.mean(alpha_1, n_1)
+        y_2 = dirichlet_multinomial.mean(alpha_2, n_2)
+        assert_equal(y[0], y_1)
+        assert_equal(y[1], y_2)
+
+    def test_lengths(self):
+        x = np.array([1, 2, 3, 4])
+        alpha = np.array([3, 4, 5])
+        n = 3
+        assert_raises(ValueError, dirichlet_multinomial.logpmf, alpha, x, n)
```

Comment:
Use `assert_raises` context manager with the `match` keyword to test the text of the message. Test all places that errors may be raised.